editor/scene_tree_dock.cpp

```diff
@@ -2896,6 +2916,42 @@ void SceneTreeDock::open_script_dialog(Node *p_for_node, bool p_extend) {
 	}
 }
 
+void SceneTreeDock::attach_shader_to_selected() {
+	if (selected_shader_material.is_null()) {
+		return;
+	}
+
+	String path = selected_shader_material->get_path();
+	if (path == "") {
+		String root_path;
+		if (editor_data->get_edited_scene_root()) {
+			root_path = editor_data->get_edited_scene_root()->get_filename();
+		}
+		String shader_name;
+		if (selected_shader_material->get_name().is_empty()) {
+			shader_name = root_path;
```

Comment:
Should be probably `root_path.get_file()`, otherwise it will try to use whole path as filename.

Reply:
Yeah, Thank you!